src/GovUk.Education.ExploreEducationStatistics.Data.Processor.Tests/Functions/ProcessorStage4ImportObservationTests.cs

```diff
@@ -206,12 +213,411 @@ await using (var contentDbContext = InMemoryContentDbContext(_contentDbContextId
+        });
+        
+        await using (var statisticsDbContext = InMemoryStatisticsDbContext(_statisticsDbContextId))
+        {
+            await statisticsDbContext.Subject.AddAsync(subject);
+            await statisticsDbContext.Filter.AddAsync(filter);
+            await statisticsDbContext.IndicatorGroup.AddAsync(indicatorGroup);
+            await statisticsDbContext.Location.AddRangeAsync(ListOf(
+                new Location
+                {
+                    GeographicLevel = GeographicLevel.LocalAuthority,
+                    Country = new Country("E92000001", "England"),
+                    LocalAuthority = new LocalAuthority("E08000025", "330", "Birmingham")
+                },
+                new Location
+                {
+                    GeographicLevel = GeographicLevel.LocalAuthority,
+                    Country = new Country("E92000001", "England"),
+                    LocalAuthority = new LocalAuthority("E08000016", "370", "Barnsley")
+                },
+                new Location
+                {
+                    GeographicLevel = GeographicLevel.LocalAuthority,
+                    Country = new Country("E92000001", "England"),
+                    LocalAuthority = new LocalAuthority("E09000011", "203", "Greenwich")
+                },
+                new Location
+                {
+                    GeographicLevel = GeographicLevel.LocalAuthority,
+                    Country = new Country("E92000001", "England"),
+                    LocalAuthority = new LocalAuthority("E09000007", "202", "Camden")
+                }));
+            await statisticsDbContext.SaveChangesAsync();
+        }
+        
+        var importObservationsMessage = new ImportObservationsMessage
+        {
+            BatchNo = 1,
+            Id = import.Id,
+            ObservationsFilePath = import.File.Path()
+        };
+
+        // There should be no interactions with BlobStorage if no batching is required.
+        var blobStorageService = new Mock<IBlobStorageService>(Strict);
+
+        var dataFilePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!,
+            "Resources" + Path.DirectorySeparatorChar + dataFileUnderTest);
+
+        var metaFilePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!,
+            "Resources" + Path.DirectorySeparatorChar + metaFileUnderTest);
+
+        blobStorageService
+            .Setup(s => s.StreamBlob(PrivateReleaseFiles, import.File.Path(), null))
+            .ReturnsAsync(() => System.IO.File.OpenRead(dataFilePath));
+
+        blobStorageService
+            .Setup(s => s.StreamBlob(PrivateReleaseFiles, import.MetaFile.Path(), null))
+            .ReturnsAsync(() => System.IO.File.OpenRead(metaFilePath));
+        
+        blobStorageService
+            .Setup(s => s.DeleteBlob(PrivateReleaseFiles, importObservationsMessage.ObservationsFilePath))
+            .Returns(Task.CompletedTask);
```

Comment:
You could use `MockBlobStorageServiceExtensions` here and in the test below, and potentially add a new extension for `StreamBlob` using a `File` or a file path?